### PR TITLE
Swagger-codegen library

### PR DESCRIPTION
**Pull request by meliaj**
Generated library from our swagger api file for v1.2.

Added in logic to handle our event types and a test for this.

Also ported over a test for handling unicode characters since this came up recently.

Major version bump.

This made me realize we had a CLI which is now gone :(

### CODE REVIEW

**Comment by meliaj** [comment]
@kvigen I updated this for 2.0. Want to give it another look?

File: clever/api_client.py [hunk 1]
```diff
@@ -121,7 +121,7 @@ def __call_api(self, resource_path, method,
             for k, v in path_params:
                 # specified safe chars, encode everything
                 resource_path = resource_path.replace(
-                    '{%s}' % k, quote(str(v), safe=config.safe_chars_for_path_param))
```
Comment (kvigen) [inline clever/api_client.py]: this isn't clever-specific, right? i wonder if we should try to fix this upstream?

Comment (meliaj) [reply clever/api_client.py]: I'm not sure this is even a real thing... But it existed before so I'm keeping up the behavior
